compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -481,6 +485,27 @@ impl<'a> From<&'a TypeCheckError> for Diagnostic {
             | TypeCheckError::InvalidShiftSize { location } => {
                 Diagnostic::simple_error(error.to_string(), String::new(), *location)
             }
+            TypeCheckError::InvalidBoolInfixOp { op, location } => {
+                let primary = match op {
+                    BinaryOpKind::Add => "Cannot add `bool` to `bool",
+                    BinaryOpKind::Subtract => "Cannot subtract `bool` from `bool",
+                    BinaryOpKind::Multiply => "Cannot multiply `bool` by `bool",
+                    BinaryOpKind::Divide => "Cannot divide `bool` by `bool`",
```

Comment:
```suggestion
                    BinaryOpKind::Add => "Cannot add a `bool` to a `bool",
                    BinaryOpKind::Subtract => "Cannot subtract a `bool` from a `bool",
                    BinaryOpKind::Multiply => "Cannot multiply a `bool` by a `bool",
                    BinaryOpKind::Divide => "Cannot divide a `bool` by a `bool`",
```

Reply:
I copied all of these messages from the Rust compiler, but I guess it's good if we try to improve on Rust :-)